src/openrct2/network/Network.cpp

```diff
@@ -1943,6 +1943,9 @@ void Network::Server_Client_Joined(const char* name, const std::string& keyhash,
         Server_Send_OBJECTS(connection, objects);
 
         // Log player joining event
+        std::string playerNameHash = player->Name + " (" + keyhash + ")";
+        player_name = (const char*)playerNameHash.c_str();
+        format_string(text, 256, STR_MULTIPLAYER_PLAYER_HAS_JOINED_THE_GAME, &player_name);
```

Comment:
You've not reserved 256 bytes of space for playerNameHash. Try calling the [reserve](https://en.cppreference.com/w/cpp/string/basic_string/reserve) method before formatting.

Reply:
`playerNameHash` isn't being written to, it's used to create the string that's passed as an argument for `format_string`. `text` is being written to, and the 256 matches its size.

I'd rather see `Util::CountOf(text)` being used instead of the hardcoded number.

Reply:
How large are these hashes, though? Is 256 characters enough for the localized text, player name, and hash added together?

Reply:
I believe the system uses SHA-1, as documented (not great as not cryptographic hash function, thus allowing collisions, but that is to be addressed later). They take 20 bytes of storage.

I do not know if the game limits players name length nor what's the macimum expected size of the localized text.

Reply:
Using repeated absolute values in the code is not great indeed, but I intended my change to be minimal.
There is clealy some refactoring to do there to drive buffer length from a single parameter, but that's out of my intended scope.

I wonder where that 256 buffer size comes from, other than loose intuition it might be enough to cover all cases. It seems the `format_string` functions collection is safe regarding too small buffers though, truncating content to null-terminate it if needed.

Reply:
From a quick check, omitting formatting tags, the longest locale string for `STR_5758` (for `STR_MULTIPLAYER_PLAYER_HAS_JOINED_THE_GAME`) is 28 bytes. Add 20 bytes for SHA-1 hash + 3 bytes for added parenthesis & space, that would leave 205 bytes for nickname, provided there is no other limit enforced anywhere else.

Aren't we on the safe side here?

Reply:
We are. @Broxzier is right — I misread the call to `format_string`. You don't have to change anything.

Reply:
I'd also add the hoststr which is the ip address as well.

Reply:
You already got that logged from some other location.
Btw, actions are accounted towards player name, not IP address, which my commit links to its SHA-1 hash, which can in turn be used for admin purposes (like banning): that was the purpose of that (minimal) change.